tracer/src/Datadog.Trace/RuntimeMetrics/RuntimeMetricsWriter.cs

```diff
@@ -233,6 +237,28 @@ private void FirstChanceException(object sender, FirstChanceExceptionEventArgs e
 
         private void GetCurrentProcessMetrics(out TimeSpan userProcessorTime, out TimeSpan systemCpuTime, out int threadCount, out long privateMemorySize)
         {
+            if (_pssConsecutiveFailures < 3 && Environment.OSVersion.Platform == PlatformID.Win32NT && Environment.OSVersion.Version > Windows81Version)
+            {
+                try
+                {
+                    ProcessSnapshotRuntimeInformation.GetCurrentProcessMetrics(out userProcessorTime, out systemCpuTime, out threadCount, out privateMemorySize);
+                    _pssConsecutiveFailures = 0;
+                }
+                catch
+                {
+                    _pssConsecutiveFailures += 1;
```

Comment:
We can technically have more than one RuntimeMetricsWriter in version conflict, right? Should we use interlocked here?

Reply:
But then they're different RuntimeMetricsWriter and they all have their own counter

Reply:
True. And we never have more than one instance of the same type, when we replace the tracer instance for example (before the old one has been disposed)?

Reply:
Though honestly I can't think of any reason _not_ to use an interlocked here so I guess I can play safe and add one